addon/lib/user.js

```diff
@@ -64,3 +68,129 @@ function makeUuid() {
 exports.getUserInfo = function () {
   return ss.storage.userInfo;
 };
+
+exports.updateProfileInfo = function (profile) {
+  if (! profile) {
+    return;
+  }
+  let currentInfo = ss.storage.profileInfo;
+  if (! currentInfo) {
+    currentInfo = ss.storage.profileInfo = {};
+  }
+  let attrs = Object.keys(profile);
+  for (let attr of attrs) {
```

Comment:
This logic is complicated...?  Could it just be:

```javascript
for (let attr in profile) {
  currentInfo[attr] = profile[attr];
}
```

?  Or are you worried about null values that don't mean erase, or...?